Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -964,8 +964,28 @@ class AnimatedInterpolation extends AnimatedWithChildren {
     super.__detach();
   }
 
+  __transformDataType(range) {
+    // Change the string array type to number array
+    // So we can reuse the same logic in iOS and Android platform
+    return range.map(function (value) {
+      if (typeof value !== 'string') {
+        return value;
+      }
+      if (/deg$/.test(value)) {
+        let degrees = parseFloat(value, 10) || 0;
+        let radians = degrees * Math.PI / 180.0;
+        return radians;
+      } else {
+        // Assume radians
+        return parseFloat(value, 10) || 0;
+      }
+    });
+  }
+
   __getNativeConfig(): any {
     NativeAnimatedHelper.validateInterpolation(this._config);
+    // Only the `outputRange` can contain strings so we don't need to tranform `inputRange` here
+    this._config.outputRange = this.__transformDataType(this._config.outputRange);
```

Comment:
Oh sorry I missed that last time I checked but can you also not mutate the original outputRange here.

You can just add it to the return object below
``` js
return {
  ...this._config,
  outputRange: this.__transformDataType(...),
  ...
}
```